web/models/user_from_auth.ex

```diff
@@ -1,48 +1,26 @@
-defmodule UserFromAuth do
-  @moduledoc """
-  Retrieve the user information from an auth request
-  """
+defmodule Dwylbot.UserFromAuth do
 
-  alias Ueberauth.Auth
+  def basic_info(auth) do
+    id = Map.get auth, :uid
+    info = Map.get auth, :info, %{}
+    avatar = Map.get info, :image
 
-  def find_or_create(%Auth{provider: :identity} = auth) do
-    case validate_pass(auth.credentials) do
-      :ok ->
-        {:ok, basic_info(auth)}
-      {:error, reason} -> {:error, reason}
-    end
-  end
-
-  def find_or_create(%Auth{} = auth) do
-    {:ok, basic_info(auth)}
-  end
-
-  defp basic_info(auth) do
-    %{id: auth.uid, name: name_from_auth(auth), avatar: auth.info.image}
+    %{id: id, name: name_from_info(info), avatar: avatar}
   end
 
-  defp name_from_auth(auth) do
-    if auth.info.name do
-      auth.info.name
-    else
-      name = [auth.info.first_name, auth.info.last_name]
+  defp name_from_info(info) do
+    case info do
+      %{name: name} when not is_nil(name) and name != "" ->
+        name
+    _ ->
+      name = [info.first_name, info.last_name]
       |> Enum.filter(&(&1 != nil and &1 != ""))
 
       cond do
-        length(name) == 0 -> auth.info.nickname
+        length(name) == 0 -> Map.get info, :nickname
```

Comment:
Again a personal thing, but more pattern matching here would be my preference for this function instead of using `case`